tests/periph_pm/main.c

```diff
@@ -0,0 +1,248 @@
+        return -1;
+    }
+
+    int mode = atoi(argv[1]);
+
+    if (mode < 0 || mode >= (int) PM_NUM_MODES) {
+        printf("Error: power mode not in range 0 - %d.\n", PM_NUM_MODES - 1);
+        return -1;
+    }
+
+    return mode;
+}
+
+static int parse_duration(int argc, char **argv)
+{
+    if (argc < 3) {
+        printf("Error: duration expected (0 - %d).\n", INT_MAX);
+        return -1;
+    }
+
+    int duration = atoi(argv[2]);
+
+    if (duration < 0) {
+        puts("Error: duration must be a positive number.");
+        return -1;
+    }
+
+    return duration;
+}
+
+#ifdef MODULE_PERIPH_RTC
+static void cb_rtc(void *arg)
+{
+    int level = (int)arg;
+
+    pm_block(level);
+}
+#endif
+
+static int cmd_block(int argc, char **argv)
+{
+    int mode = parse_mode(argc, argv);
+
+    if (mode < 0) {
+        return 1;
+    }
+
+    printf("Blocking power mode %d.\n", mode);
+    fflush(stdout);
+
+    pm_block(mode);
+
+    return 0;
+}
+#endif
+
+static int cmd_lowest(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+
+    puts("CPU will enter lowest power mode possible.");
+    fflush(stdout);
+
+    pm_set_lowest();
+
+    return 0;
+}
+
+static int cmd_off(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+
+    puts("CPU will turn off.");
+    fflush(stdout);
+
+    pm_off();
+
+    return 0;
+}
+
+static int cmd_reboot(int argc, char **argv)
+{
+    (void) argc;
+    (void) argv;
+
+    puts("CPU will reboot.");
+    fflush(stdout);
+
+    pm_reboot();
+
+    return 0;
+}
+
+#ifdef MODULE_PM_LAYERED
+static int cmd_set(int argc, char **argv)
+{
+    int mode = parse_mode(argc, argv);
+
+    if (mode < 0) {
+        return 1;
+    }
+
+    printf("CPU will enter power mode %d.\n", mode);
+    fflush(stdout);
+
+    pm_set(mode);
+
+    return 0;
+}
+
+static int cmd_unblock(int argc, char **argv)
+{
+    int mode = parse_mode(argc, argv);
+
+    if (mode < 0) {
+        return 1;
+    }
+
+    printf("Unblocking power mode %d.\n", mode);
+    fflush(stdout);
+
+    pm_unblock(mode);
+
+    return 0;
+}
+
+#ifdef MODULE_PERIPH_RTC
+static int cmd_unblock_rtc(int argc, char **argv)
+{
+    struct tm time;
+
+    int mode = parse_mode(argc, argv);
+    int duration = parse_duration(argc, argv);
+
+    if (mode < 0 || duration < 0) {
+        return 1;
+    }
+
+    printf("Unblocking power mode %d for %d seconds.\n", mode, duration);
+    fflush(stdout);
+
+    rtc_get_time(&time);
+    time.tm_sec += duration;
+    while (time.tm_sec > 60) {
+        time.tm_min++;
+        time.tm_sec -= 60;
+    }
+    while (time.tm_min > 60) {
+        time.tm_hour++;
+        time.tm_min -= 60;
+    }
+    rtc_set_alarm(&time, cb_rtc, (void *)mode);
+
+    pm_unblock(mode);
+
+    return 0;
+}
+#endif
+#endif
+
+int main(void)
+{
+    char line_buf[SHELL_DEFAULT_BUFSIZE];
+
+    /* print some information about the modes */
+    printf("This application allows you to test the power management\n"
```

Comment:
Not if power management is a peripheral. I would just say `to test the CPU power management`

Reply:
Fixed.